File: dp_primer/dp_primer_partial_sums_step0.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] nx = ReadLine().Split( ' ' ).Select( int.Parse ).ToArray();
        int[] a = new int[nx[0]];
        bool[] p = new bool[nx[1] + 1];

        foreach( int i in Range( 0, nx[0] ) ) {
            a[i] = int.Parse( ReadLine() );
        }
        p[0] = true;
        foreach( int i in Range( 0, nx[0] ) ) {
            for( int j = nx[1]; a[i] <= j; j-- ) {
                if( p[j - a[i]] == true ) {
                    p[j] = true;
                }
            }
        }
        WriteLine( p[nx[1]] ? "yes" : "no" );
        return;
    }
}

File: dp_primer/dp_primer_partial_sums_step0.py
# 実行時間 30ms
n, x = list(map(int, input().split()))
a = []
p = [False for _ in [0] * (x + 1)]

for _ in range(n):
    a.append(int(input()))
p[0] = True
for i in range(n):
    for j in range(x, a[i] - 1, -1):
        if p[j - a[i]] == True:
            p[j] = True
print("yes" if p[x] else "no")
